packages/gatsby-theme-docs/src/layouts/content.js

```diff
@@ -24,7 +24,8 @@ import PageContentInset from './internals/page-content-inset';
 import PageReadTime from './internals/page-read-time-estimation';
 
 const LayoutContent = (props) => {
-  const { ref, inView } = useInView();
+  const { ref, inView, entry } = useInView();
+  const isSearchBoxInView = !Boolean(entry) || inView;
```

Comment:
This fixes the issue that the search input in the page nav initially appears on page load and then disappears.
The `entry` is initially undefined so we can leverage that to fix the behavior.